wire-ios-sync-engine/Tests/Source/Synchronization/Strategies/CallingRequestStrategyTests.swift

```diff
@@ -161,319 +161,340 @@ class CallingRequestStrategyTests: MessagingTest {
-        XCTAssertNotNil(request)
-        XCTAssertEqual(request?.path, "/conversations/\(conversation.remoteIdentifier!.transportString())/otr/messages")
+        syncMOC.performAndWait {
+            let request = sut.nextRequest(for: .v0)
+            XCTAssertNotNil(request)
+            XCTAssertEqual(request?.path, "/conversations/\(conversation.remoteIdentifier!.transportString())/otr/messages")
 
-        // When
-        request?.complete(with: ZMTransportResponse(payload: payload as ZMTransportData, httpStatus: 412, transportSessionError: nil, apiVersion: APIVersion.v0.rawValue))
+            // When
+            request?.complete(with: ZMTransportResponse(payload: payload as ZMTransportData, httpStatus: 412, transportSessionError: nil, apiVersion: APIVersion.v0.rawValue))
+        }
 
         // Then
         XCTAssertTrue(waitForCustomExpectations(withTimeout: 0.5))
     }
 
     func testThatItGeneratesClientListRequestAndCallsTheCompletionHandler_Federated() throws {
-        // Given
-        BackendInfo.isFederationEnabled = true
-
-        let selfClient = createSelfClient()
-        let selfUser = ZMUser.selfUser(in: syncMOC)
-        selfUser.domain = "foo.com"
-
-        // One user with two clients connected to self.
-        let user1 = ZMUser.insertNewObject(in: syncMOC)
-        user1.remoteIdentifier = .create()
-        user1.domain = "foo.com"
-        let client1 = createClient(for: user1, connectedTo: selfClient)
-        let client2 = createClient(for: user1, connectedTo: selfClient)
-
-        // Another user with two clients connected to self.
-        let user2 = ZMUser.insertNewObject(in: syncMOC)
-        user2.remoteIdentifier = .create()
-        user2.domain = "bar.com"
-        let client3 = createClient(for: user2, connectedTo: selfClient)
-        let client4 = createClient(for: user2, connectedTo: selfClient)
-
-        // A conversation with both users and self.
-        let conversation = ZMConversation.insertNewObject(in: syncMOC)
-        conversation.remoteIdentifier = .create()
-        conversation.domain = "foo.com"
-        conversation.messageProtocol = .proteus
-        conversation.addParticipantsAndUpdateConversationState(
-            users: [selfUser, user1, user2],
-            role: nil
-        )
-
-        conversation.needsToBeUpdatedFromBackend = false
-        syncMOC.saveOrRollback()
-
-        let payload = """
-        {
-            "missing": {
-                "foo.com": {
-                    "\(user1.remoteIdentifier.uuidString)": ["\(client1.remoteIdentifier!)", "\(client2.remoteIdentifier!)"]
-                },
-                "bar.com": {
-                    "\(user2.remoteIdentifier.uuidString)": ["\(client3.remoteIdentifier!)", "\(client4.remoteIdentifier!)"]
-                }
-            }
-        }
-        """
+        let (conversation, payload) = try syncMOC.performAndWait {
+            // Given
+            BackendInfo.isFederationEnabled = true
```

Comment:
do we change the BackendInfo.storage here ?
something like `BackendInfo.storage = .random()!`
in order not to affect other tests

Reply:
it can be done later - not a blocker